### PR TITLE
Part 3: Create new 'bittensor-sdk` package

### PR DESCRIPTION
**Pull request by basfroman**
- removing obsolete, unused (community and bittensor) code
- move backwards compatibility related content to `bittensor/utils/backwards_compatibility` subpackage

### CODE REVIEW

File: bittensor/utils/backwards_compatibility/subnets.py [hunk 1]
```diff
@@ -16,16 +16,18 @@
...
-
-from bittensor_wallet import Wallet
+from typing import Any, List, Union, Optional, TYPE_CHECKING
 
 from bittensor.core.axon import Axon
 from bittensor.core.dendrite import Dendrite
-from bittensor.core.synapse import Synapse
 from bittensor.utils.btlogging import logging
 
+if TYPE_CHECKING:
```
Comment (thealligatorking) [inline bittensor/utils/backwards_compatibility/subnets.py:26]: Can you add a comment to explain here.

**Review by thealligatorking — APPROVED** [review]

File: bittensor/core/dendrite.py [hunk 2]
```diff
@@ -784,25 +757,21 @@ async def __aenter__(self):
...
+        Enables the use of the ``async with`` statement with the Dendrite instance. When entering the context, the current instance of the class is returned, making it accessible within the asynchronous context.
 
         Returns:
             Dendrite: The current instance of the Dendrite class.
 
         Usage::
-
-            async with Dendrite() as dendrite:
-                await dendrite.some_async_method()
+            async with Dendrite() as dendrite: await dendrite.some_async_method()
```
Comment (thewhaleking) [inline bittensor/core/dendrite.py]: Why is this changed?

File: bittensor/core/dendrite.py [hunk 3]
```diff
@@ -811,8 +780,7 @@ async def __aexit__(self, exc_type, exc_value, traceback):
 
         Usage::
 
-            async with bt.dendrite( wallet ) as dendrite:
-                await dendrite.some_async_method()
+            async with bt.dendrite( wallet ) as dendrite: await dendrite.some_async_method()
```
Comment (thewhaleking) [inline bittensor/core/dendrite.py]: Why is this changed?

**Review by thewhaleking — APPROVED** [review]

File: bittensor/core/dendrite.py (hunk 2 above)
Comment (basfroman) [reply bittensor/core/dendrite.py]: Accidentally. Good catch! 
Fixed back.

File: bittensor/core/dendrite.py (hunk 3 above)
Comment (basfroman) [reply bittensor/core/dendrite.py]: Accidentally. Good catch too! 
Fixed back.

File: bittensor/utils/backwards_compatibility/subnets.py (hunk 1 above)
Comment (basfroman) [reply bittensor/utils/backwards_compatibility/subnets.py:26]: Yep. Done.
